app/team.go

```diff
@@ -632,6 +632,15 @@ func (a *App) JoinUserToTeam(team *model.Team, user *model.User, userRequestorId
 		return err
 	}
 
+	if err := a.createInitialSidebarCategories(user, team); err != nil {
+		mlog.Error(
+			"Encountered an issue creating default sidebar categories.",
+			mlog.String("user_id", user.Id),
+			mlog.String("team_id", team.Id),
+			mlog.Err(err),
+		)
```

Comment:
Should we return from here or not ?

Reply:
Good question. @hmhealey - pretty sure we can't allow to join the team if something fails here...

Reply:
We unfortunately don't have a good pattern of handling errors in side effects like this. Right below this, it tries to join the Town Square and Off Topic channels, but if that fails, it just logs an error and continues.

Since we don't return for other more-critical errors, I don't think we should return here either.